### PR TITLE
Update the FreeRTOS-Kernel submodule to V10.6.1

### PR DESCRIPTION
**Pull request by Skptak**
Description
-----------
Update the FreeRTOS-Kernel submodule to V10.6.1
Add the manifest verification to the workflow as well

Test Steps
-----------

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: manifest.yml [hunk 1]
```diff
@@ -1,6 +1,8 @@
-name : "FreeRTOS-Plus-TCP"
-version: "V4.0.0"
-description: "Thread safe FreeRTOS TCP/IP stack working on top of the FreeRTOS-Kernel to implement the TCP/IP protocol. Suitable for microcontrollers."
+name: "FreeRTOS-Plus-TCP"
+version: "V4.0.0-rc3"
```
Comment (amazonKamath) [inline manifest.yml:2]: V4.0.0?

**Review by amazonKamath — APPROVED** [review]

File: manifest.yml (hunk 1 above)
Comment (moninom1) [inline manifest.yml:2]: Yeah it should be "V4.0.0"

**Comment by chinglee-iot** [comment]
FreeRTOS-Kernel/ThirdParty/GCC/Posix/portmacro.h is used before ConfigFiles/portmacro.h.
I think this causes xMPU_SETTINGS is undefined since it is defined in ConfigFiles/portmacro.h.
We can adjust the include order to see if the CI problem is solved.

The processed output of mock_task.c.
```
# 1 "/mnt/c/msys64/home/chinglee/plus_tcp/FreeRTOS-Plus-TCP/test/FreeRTOS-Kernel/portable/ThirdParty/GCC/Posix/portmacro.h" 1
# 31 "/mnt/c/msys64/home/chinglee/plus_tcp/FreeRTOS-Plus-TCP/test/FreeRTOS-Kernel/portable/ThirdParty/GCC/Posix/portmacro.h"
#define PORTMACRO_H 
```

**Pull request closed without merging**